### PR TITLE
Benchmarks

### PR DESCRIPTION
**Pull request by shaunxw**

### CODE REVIEW

File: benches/helpers.rs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            tokio::time::sleep(SLOW_CALL).await;
+            Result::<_, jsonrpsee::core::Error>::Ok("0x42")
+        })
+        .unwrap();
+
+    module
+}
+
+pub mod fixed_client {
+    use jsonrpsee_v0_15::client_transport::ws::{Uri, WsTransportClientBuilder};
```
Comment (xlc) [inline benches/helpers.rs:126]: why is this needed? does this mean the 0.18 version is buggy?

File: src/server.rs [hunk 2]
```diff
@@ -87,11 +87,17 @@ pub async fn start_server(
...
             let middlewares = middlewares.clone();
             async move {
-                let params = params
-                    .parse::<JsonValue>()?
-                    .as_array()
-                    .ok_or_else(|| CallError::InvalidParams(anyhow::Error::msg("invalid params")))?
-                    .to_owned();
+                let parsed = params.parse::<JsonValue>()?;
+                let params = if parsed == JsonValue::Null {
+                    vec![]
```
Comment (xlc) [inline src/server.rs:92]: need some tests

**Review by xlc — COMMENTED** [review]
can you update readme to indicate how to run the benchmarks

**Comment by shaunxw** [comment]
updated with params parse unit tests the non-fixed jsonrpsee websocket client works now.

**Review by xlc — APPROVED** [review]
